test/simulator_tests/birth_death_simulator_test.py

```diff
@@ -0,0 +1,573 @@
+        tree = nx.balanced_tree(2, 3, create_using=nx.DiGraph())
+        leaves = get_leaves(tree)
+        for i in leaves:
+            bd_sim.remove_and_prune_lineage(i, tree)
+
+        self.assertEqual(list(tree.edges), [])
+
+    def test_remove_and_prune_lineage_1(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        tree = nx.balanced_tree(2, 3, create_using=nx.DiGraph())
+        bd_sim.remove_and_prune_lineage(11, tree)
+        bd_sim.remove_and_prune_lineage(13, tree)
+        bd_sim.remove_and_prune_lineage(14, tree)
+
+        expected_edges = [
+            (0, 1),
+            (0, 2),
+            (1, 3),
+            (1, 4),
+            (2, 5),
+            (3, 7),
+            (3, 8),
+            (4, 9),
+            (4, 10),
+            (5, 12),
+        ]
+        self.assertEqual(list(tree.edges), expected_edges)
+
+    def test_remove_and_prune_lineage_2(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        tree = nx.balanced_tree(2, 3, create_using=nx.DiGraph())
+        for i in range(7, 11):
+            bd_sim.remove_and_prune_lineage(i, tree)
+
+        expected_edges = [
+            (0, 2),
+            (2, 5),
+            (2, 6),
+            (5, 11),
+            (5, 12),
+            (6, 13),
+            (6, 14),
+        ]
+        self.assertEqual(list(tree.edges), expected_edges)
+
+    def test_collapse_unifurcations_source(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        tree = nx.DiGraph()
+        tree.add_nodes_from(list(range(6)))
+        tree.add_edges_from([(0, 1), (1, 2), (2, 3), (2, 4), (3, 5)])
+        for i in tree.edges:
+            tree.edges[i]["weight"] = 1.5
+
+        bd_sim.collapse_unifurcations(tree, source=1)
+
+        expected_edges = [
+            (0, 1, {"weight": 1.5}),
+            (1, 4, {"weight": 3.0}),
+            (1, 5, {"weight": 4.5}),
+        ]
+        self.assertEqual(list(tree.edges(data=True)), expected_edges)
+
+    def test_collapse_unifurcations_1(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        tree = nx.DiGraph()
+        tree.add_nodes_from(list(range(10)))
+        tree.add_edges_from(
+            [
+                (0, 1),
+                (0, 2),
+                (2, 3),
+                (3, 4),
+                (2, 5),
+                (5, 6),
+                (6, 7),
+                (6, 8),
+                (2, 9),
+            ]
+        )
+        for i in tree.edges:
+            tree.edges[i]["weight"] = 1.5
+
+        bd_sim.collapse_unifurcations(tree)
+        expected_edges = [
+            (0, 1, {"weight": 1.5}),
+            (0, 2, {"weight": 1.5}),
+            (2, 9, {"weight": 1.5}),
+            (2, 4, {"weight": 3.0}),
+            (2, 6, {"weight": 3.0}),
+            (6, 7, {"weight": 1.5}),
+            (6, 8, {"weight": 1.5}),
+        ]
+        self.assertEqual(list(tree.edges(data=True)), expected_edges)
+
+    def test_collapse_unifurcations_2(self):
```

Comment:
Can we be more descriptive with our names? What exactly is this test doing?

Reply:
Yeah, generally, I think we should document all our tests.